packages/core/src/views/connect/Index.svelte

```diff
@@ -463,7 +558,7 @@
             {/if}
           {/if}
 
-          {#if $modalStep$ === 'connectingWallet' && selectedWallet}
+          {#if ($modalStep$ === 'connectingWallet' && selectedWallet && windowWidth >= MOBILE_WINDOW_WIDTH) || (windowWidth <= MOBILE_WINDOW_WIDTH && connectionRejected && $modalStep$ === 'connectingWallet' && selectedWallet)}
```

Comment:
This logic seems pretty long, can it be prettier-ified or broke into a function?